src/library/scanner/recursivescandirectorytask.cpp

```diff
@@ -54,7 +56,7 @@ void RecursiveScanDirectoryTask::run() {
         if (currentFileInfo.isFile()) {
             const QString& fileName = currentFileInfo.fileName();
             if (supportedExtensionsRegex.indexIn(fileName) != -1) {
-                newHashStr.append(currentFile);
+                hasher.addData(currentFile.toUtf8());
```

Comment:
Did you made a performance check for toUtf8() compared to use the QString bytes? 
I can Imagine that the later is faster.

Reply:
Raw bytes of QString are not platform-independent when considering endianess, but UTF-8 is. Does this matter?

Reply:
There is no function to obtain the raw bytes of a QString. Using toLatin1() may lose characters an toLocal8Bit() is platform dependent while the database is portable. Please clarify which option you consider here.

Reply:
It was just the idea to check if something like 
hasher.addData(reinterprete_cast<char*>(currentFile.utf16()))
is faster or not. 
This way there is the double amount to hash but no utf8 conversion. 

I don't care about endianes, because this would be only apply if the DB is on an external HD, accessing the same OS with different CPU architectures.
Thinking of this a independent solution is not bad.

Reply:
I don't think that the additional allocations really matters here and won't invest hours to figure out how to do a realistic performance test to prove this. The optimized solution would also produce non-portable data, in contrast to all other data in the database.

Reply:
I'm not sure these micro-optimizations are worth it., but we could also use `currentFile.constData()`. If I understand the docs correctly it should be utf-8 as well since QString uses it internally:

> QString::QString(const char *str)
Constructs a string initialized with the 8-bit string str. The given const char pointer is converted to Unicode using the fromUtf8() function.

Reply:
Sorry, just ignore my previous comment. I forgot that constData returns qchar, not char.

Reply:
> It was just the idea to check if something like
> hasher.addData(reinterprete_cast<char*>(currentFile.utf16()))

Double the data should not be an issue. sha256 runtime not increase 1:1 with the amount of input data. I'm on my phone, but you should see the same effect on desktop pcs:

```
$ openssl speed sha256
The 'numbers' are in 1000s of bytes per second processed.
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes  16384 bytes
sha256          180714.92k   532643.80k  1088833.54k  1466255.70k  1633716.91k  1647209.13k
```

Reply:
The actual number of bytes that we feed into QCryptographicHash should not be an issue. But QString might need to allocate a temporary QByteArray for the UTF-8 representation and the conversion itself also takes some time.

Yes, utf16() would be faster, but it is non-portable. If someone is able to proof that it is much faster and worth the drawback, please go ahead ;) I will not do it.

Reply:
Just out of curious I did the test. Here are the results: 
```
10 x 30 characters
  6040 nsec hasher.addData(reinterpret_cast<const char*>(lorem.unicode()), lorem.size() * sizeof(QChar));
  8683 nsec hasher.addData(lorem.toUtf8());
 12836 nsec hasher.addData(lorem.toUtf8()); "ööööö.."

10 x 888 characters 
149882 nsec hasher.addData(reinterpret_cast<const char*>(lorem.unicode()), lorem.size() * sizeof(QChar));
 82302 nsec hasher.addData(lorem.toUtf8());
257479 nsec hasher.addData(lorem.toUtf8()); "ööööö.."
```

Conclusion: 
As expected there is only performance gain for ASCII characters. Except in non Latin charter sets two byte utf8 characters are a minority. 
For a typical file name of 30 charters, the unicode() version is way faster. If we append all strings first, the utf8 version catches up at one point.

Reply:
Those numbers also need to be put in relation to the file system access. I guess that the differences on the overall performance are negligible. That's what I had in mind with "realistic performance test".

Reply:
Yes of cause. I was just interested if we have a low hanging fruit here.